openmdao/components/tests/test_submodel_comp.py

```diff
@@ -0,0 +1,372 @@
+                self._get_subsystem('subprob2').add_output('y')
+
+                self.promotes('subprob2', ['r'])
+                self.promotes('subprob2', ['theta'])
+                self.promotes('subprob2', ['y'])
+
+        p = om.Problem()
+
+        model = om.Group()
+        model.add_subsystem('supComp', om.ExecComp('z = x**2 + y'),
+                            promotes_inputs=['x', 'y'],
+                            promotes_outputs=['z'])
+
+        p.model.add_subsystem('sub1', Subsys1(), promotes_inputs=['r', 'theta'],
+                              promotes_outputs=['x'])
+        p.model.add_subsystem('sub2', Subsys2(), promotes_inputs=['r', 'theta'],
+                              promotes_outputs=['y'])
+        p.model.add_subsystem('supModel', model, promotes_inputs=['x','y'],
+                              promotes_outputs=['z'])
+
+        p.setup(force_alloc_complex=True)
+
+        p.set_val('r', 1)
+        p.set_val('theta', pi)
+
+        p.run_model()
+        cpd = p.check_partials(method='cs', out_stream=None)
+
+        assert_near_equal(p.get_val('z'), 1)
+
+    def test_invalid_io(self):
+        p = om.Problem()
+
+        model = om.Group()
+        model.add_subsystem('supComp', om.ExecComp('z = x**2 + y'),
+                            promotes_inputs=['x', 'y'],
+                            promotes_outputs=['z'])
+
+        submodel1 = om.Group()
+        submodel1.add_subsystem('sub1_ivc_r', om.IndepVarComp('r', 1.),
+                                promotes_outputs=['r'])
+        submodel1.add_subsystem('sub1_ivc_theta', om.IndepVarComp('theta', pi),
+                                promotes_outputs=['theta'])
+        submodel1.add_subsystem('subComp1', om.ExecComp('x = r*cos(theta)'),
+                                promotes_inputs=['r', 'theta'],
+                                promotes_outputs=['x'])
+
+        submodel2 = om.Group()
+        submodel2.add_subsystem('sub2_ivc_r', om.IndepVarComp('r', 2),
+                                promotes_outputs=['r'])
+        submodel2.add_subsystem('sub2_ivc_theta', om.IndepVarComp('theta', pi/2),
+                                promotes_outputs=['theta'])
+        submodel2.add_subsystem('subComp2', om.ExecComp('y = r*sin(theta)'),
+                                promotes_inputs=['r', 'theta'],
+                                promotes_outputs=['y'])
+
+        subprob1 = om.SubmodelComp(model=submodel1)
+        subprob2 = om.SubmodelComp(model=submodel2)
+
+        subprob1.add_input('psi')
+
+        p.model.add_subsystem('sub1', subprob1)
+        p.model.add_subsystem('sub2', subprob2)
+        p.model.add_subsystem('supModel', model, promotes_inputs=['x','y'],
+                              promotes_outputs=['z'])
+
+        with self.assertRaises(Exception) as ctx:
+            p.setup(force_alloc_complex=True)
+            
+        msg = 'Variable psi does not exist in model.'
+        self.assertTrue(str(ctx.exception), msg)
```

Comment:
You need an `assertEqual` here to check that the exception matches the message.

Reply:
changed to `assertEqual`